Plugin crashes with `unexpected EOF` despite valid config

**Description**

Running the plugin results in a crash with unexpected EOF, even with a valid configuration. The plugin process exits with status 2.

**Configuration**
```
loglevel: trace
write-directory: evaluation_results
write: true
output: json
services:
  hashbrown:
    plugin: github-repo

    policy:
      catalogs:
        - OSPS_B
      applicability:
        - Maturity Level 1

    vars:
      owner: rust-lang
      repo: hashbrown
      token: <redacted>
```

**Output**
```
./privateer run
2025/04/29 15:20:10 vars: map[]
2025-04-29T15:20:18.432+0200 [ERROR] tests failed in plugin hashbrown: unexpected EOF
2025-04-29T15:20:18.432+0200 [ERROR] plugin process exited: plugin=/.privateer/bin/github-repo id=447337 error="exit status 2
```

**Expected Behavior**
The plugin should be creating `hashbrown.json` under `evaluation_results/hashbrown`

**Actual Behavior**
Log file (`hashbrown.log`) is created, but json (`hashbrown.json`) is missing

Also attaching  [hashbrown.log](https://github.com/user-attachments/files/19959779/hashbrown.log) file

Continuing to debug... I can use the same config with three different repos, and only the `hashbrown` repo fails.

Current theory is that it might be something related to the payload data being different in that repo for some reason. Gotta investigate to see where there might be potential for deltas.

<details><summary>Config used</summary>

```yaml
loglevel: error # increase to trace for more verbose output
write-directory: evaluation_results
write: true # Change to false if you don't want to write the results to disk
output: json # Change to json if you want CLI and written output in json format

vars:
  token: <redacted>

services:

  flux2: # unique name/id for the evaluation
    plugin: github-repo # required while running via Privateer
    policy:
      catalogs:
        - OSPS_B # currently this is the only available catalog to assess against
      applicability:
        - Maturity Level 1
        - Maturity Level 2
        - Maturity Level 3
    vars:
      owner: fluxcd
      repo: flux2

  pvtr: # unique name/id for the evaluation
    plugin: github-repo # required while running via Privateer
    policy:
      catalogs:
        - OSPS_B # currently this is the only available catalog to assess against
      applicability:
        - Maturity Level 1
        - Maturity Level 2
        - Maturity Level 3
    vars:
      owner: privateerproj
      repo: privateer

  hashbrown:
    plugin: github-repo

    policy:
      catalogs:
        - OSPS_B
      applicability:
        - Maturity Level 1

    vars:
      owner: rust-lang
      repo: hashbrown
```

</details>

<details>
<summary>
Successful run 1

`$ go run . debug --service=flux2`

</summary>

```sh
Running in debug mode
2025/04/30 13:27:01 vars: map[owner:fluxcd repo:flux2 token:<redacted>]
{"@level":"error","@message":"OSPS-AC-01.01: Two-factor authentication is NOT configured as required by the parent organization","@timestamp":"2025-04-30T13:27:25.925263-05:00"}
{"@level":"error","@message":"OSPS-AC-04.01: Workflow permissions default to ","@timestamp":"2025-04-30T13:27:25.925520-05:00"}
{"@level":"error","@message":"OSPS-BR-01.01: Error parsing workflow: [:22:0: could not parse as YAML: yaml: line 22: mapping values are not allowed in this context [syntax-check]]","@timestamp":"2025-04-30T13:27:25.926726-05:00"}
{"@level":"error","@message":"OSPS-BR-03.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:27:25.926897-05:00"}
{"@level":"error","@message":"OSPS-BR-06.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:27:25.926924-05:00"}
{"@level":"error","@message":"OSPS-DO-01.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:27:25.926937-05:00"}
{"@level":"error","@message":"OSPS-DO-02.01: Repository does not accept vulnerability reports","@timestamp":"2025-04-30T13:27:25.928455-05:00"}
{"@level":"error","@message":"OSPS-DO-03.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:27:25.928500-05:00"}
{"@level":"error","@message":"OSPS-DO-04.01: A support.md file or support statements in the readme.md was NOT found","@timestamp":"2025-04-30T13:27:26.090334-05:00"}
{"@level":"error","@message":"OSPS-DO-05.01: A support.md file or support statements in the readme.md was NOT found","@timestamp":"2025-04-30T13:27:26.221730-05:00"}
{"@level":"error","@message":"OSPS-DO-06.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:27:26.221906-05:00"}
{"@level":"error","@message":"OSPS-GV-01.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:27:26.222054-05:00"}
{"@level":"error","@message":"OSPS-GV-03.02: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:27:26.222163-05:00"}
{"@level":"error","@message":"OSPS-QA-03.01: Some executed status checks are not mandatory but all should be: DCO, scan-fossa, scan-snyk, scan-codeql, Snyk Open Source, CodeQL, pull-request","@timestamp":"2025-04-30T13:27:26.426910-05:00"}
{"@level":"error","@message":"OSPS-QA-04.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:27:26.427160-05:00"}
{"@level":"error","@message":"OSPS-QA-05.01: Suspected binaries found in the repository: DCO, MAINTAINERS","@timestamp":"2025-04-30T13:27:26.427251-05:00"}
{"@level":"error","@message":"OSPS-QA-07.01: Branch protection rule does not require reviews","@timestamp":"2025-04-30T13:27:26.427348-05:00"}
{"@level":"error","@message":"OSPS-VM-02.01: Security contacts were not specified in Security Insights data","@timestamp":"2025-04-30T13:27:26.427435-05:00"}
{"@level":"error","@message":"OSPS-VM-06.01: No dependency management file found","@timestamp":"2025-04-30T13:27:26.427571-05:00"}
{"@level":"error","@message":"\u003e flux2_OSPS_B: 7 Passed, 13 Warnings, 19 Failed","@timestamp":"2025-04-30T13:27:26.427738-05:00"}
```

</details>


<details>
<summary>
Successful run 2:

`$ go run . debug --service=pvtr`

</summary>

```sh
Running in debug mode
2025/04/30 13:26:32 vars: map[owner:privateerproj repo:privateer token:<redacted>]
{"@level":"error","@message":"OSPS-AC-01.01: Two-factor authentication is NOT configured as required by the parent organization","@timestamp":"2025-04-30T13:26:44.867211-05:00"}
{"@level":"error","@message":"OSPS-BR-06.01: No SLSA attestation found in security insights","@timestamp":"2025-04-30T13:26:44.870722-05:00"}
{"@level":"error","@message":"OSPS-DO-03.01: Signature verification guide was NOT specified in Security Insights data","@timestamp":"2025-04-30T13:26:44.870841-05:00"}
{"@level":"error","@message":"OSPS-DO-04.01: A support.md file or support statements in the readme.md was NOT found","@timestamp":"2025-04-30T13:26:44.979349-05:00"}
{"@level":"error","@message":"OSPS-DO-05.01: A support.md file or support statements in the readme.md was NOT found","@timestamp":"2025-04-30T13:26:45.131876-05:00"}
{"@level":"error","@message":"OSPS-DO-06.01: Dependency management policy was NOT specified in Security Insights data","@timestamp":"2025-04-30T13:26:45.132052-05:00"}
{"@level":"error","@message":"OSPS-GV-01.02: Roles and responsibilities were NOT specified in Security Insights data","@timestamp":"2025-04-30T13:26:45.132449-05:00"}
{"@level":"error","@message":"OSPS-GV-03.02: Code review guide was NOT specified in Security Insights data","@timestamp":"2025-04-30T13:26:45.132555-05:00"}
{"@level":"error","@message":"OSPS-QA-03.01: Some executed status checks are not mandatory but all should be: Acceptance Tests, release / create_release, goreleaser (NOTE: Not continuing to evaluate branch protection: combining requirements in rulesets and branch protection is not recommended)","@timestamp":"2025-04-30T13:26:45.296344-05:00"}
{"@level":"error","@message":"OSPS-QA-07.01: Branch protection rule requires 0 approving reviews","@timestamp":"2025-04-30T13:26:45.296605-05:00"}
{"@level":"error","@message":"OSPS-VM-06.01: No dependency management file found","@timestamp":"2025-04-30T13:26:45.296760-05:00"}
{"@level":"error","@message":"\u003e pvtr_OSPS_B: 15 Passed, 13 Warnings, 11 Failed","@timestamp":"2025-04-30T13:26:45.296840-05:00"}
```

</details>


<details>
<summary>
Failed run:

`$ go run . debug --service=hashbrown`

</summary>

```sh
Running in debug mode
2025/04/30 13:28:04 vars: map[owner:rust-lang repo:hashbrown token:<redacted>]
{"@level":"error","@message":"OSPS-AC-01.01: Two-factor authentication is NOT configured as required by the parent organization","@timestamp":"2025-04-30T13:28:12.354121-05:00"}
{"@level":"error","@message":"OSPS-BR-03.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:28:12.356106-05:00"}
{"@level":"error","@message":"OSPS-DO-01.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:28:12.356393-05:00"}
{"@level":"error","@message":"OSPS-DO-02.01: Repository does not accept vulnerability reports","@timestamp":"2025-04-30T13:28:12.356660-05:00"}
{"@level":"error","@message":"OSPS-GV-03.01: Contribution guide not found in Security Insights data or via GitHub API","@timestamp":"2025-04-30T13:28:12.356929-05:00"}
{"@level":"error","@message":"OSPS-QA-04.01: Security insights required for this assessment, but file not found","@timestamp":"2025-04-30T13:28:12.434330-05:00"}
{"@level":"error","@message":"OSPS-QA-05.01: Suspected binaries found in the repository: LICENSE-APACHE, LICENSE-MIT","@timestamp":"2025-04-30T13:28:12.434389-05:00"}
{"@level":"error","@message":"OSPS-VM-02.01: Security contacts were not specified in Security Insights data","@timestamp":"2025-04-30T13:28:12.434421-05:00"}
{"@level":"error","@message":"\u003e hashbrown_OSPS_B: 5 Passed, 3 Warnings, 8 Failed","@timestamp":"2025-04-30T13:28:12.434454-05:00"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x100c91124]

goroutine 1 [running]:
encoding/json.(*encodeState).marshal.func1()
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:294 +0x80
panic({0x1013eb460?, 0x101b2b6b0?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/runtime/panic.go:785 +0x124
net/url.(*URL).String(0x1013fc620?)
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/net/url/url.go:832 +0x24
github.com/shurcooL/githubv4.URI.MarshalJSON({0x16?})
        /Users/knight/go/pkg/mod/github.com/shurcoo!l/githubv4@v0.0.0-20240727222349-48295856cce7/scalar.go:80 +0x1c
encoding/json.addrMarshalerEncoder(0x14000272580, {0x101503b60?, 0x14000184950?, 0x10143d660?}, {0x18?, 0xb0?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:460 +0x150
encoding/json.condAddrEncoder.encode({0x10153de10?, 0x10153de58?}, 0x140000db018?, {0x101503b60?, 0x14000184950?, 0x100c59c54?}, {0xc0?, 0x26?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:902 +0x50
encoding/json.structEncoder.encode({{{0x14000376240, 0x2, 0x2}, 0x140003a0b40, 0x140003a0b70}}, 0x14000272580, {0x10143d6e0?, 0x14000184940?, 0x1013926c0?}, {0x19?, ...})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:715 +0x190
encoding/json.structEncoder.encode({{{0x1400050bb08, 0xb, 0x10}, 0x140003a1590, 0x140003a15c0}}, 0x14000272580, {0x1014fac80?, 0x14000184840?, 0x140000db258?}, {0xd?, ...})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:715 +0x190
encoding/json.structEncoder.encode({{{0x1400050b208, 0x9, 0x10}, 0x140000bd980, 0x140000bd9b0}}, 0x14000272580, {0x1015063e0?, 0x140000540c0?, 0x140000db358?}, {0x60?, ...})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:715 +0x190
encoding/json.(*encodeState).reflectValue(0x14000272580, {0x1015063e0?, 0x140000540c0?, 0x16?}, {0x18?, 0xb4?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:322 +0x70
encoding/json.interfaceEncoder(0x14000272580, {0x1013c4ac0?, 0x1400026a790?, 0x100c578c8?}, {0x60?, 0xba?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:669 +0xcc
encoding/json.ptrEncoder.encode({0x140000db508?}, 0x14000272580, {0x1013688a0?, 0x1400026a790?, 0x140000db518?}, {0x60?, 0x7f?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:887 +0x1dc
encoding/json.(*encodeState).reflectValue(0x14000272580, {0x1013688a0?, 0x1400026a790?, 0x100c59c70?}, {0xb8?, 0xb5?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:322 +0x70
encoding/json.interfaceEncoder(0x14000272580, {0x1013c4ac0?, 0x140001bf580?, 0x100c578c8?}, {0x18?, 0xb6?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:669 +0xcc
encoding/json.structEncoder.encode({{{0x14000182248, 0x4, 0x4}, 0x1400024b440, 0x1400024b470}}, 0x14000272580, {0x1014e7240?, 0x140001bf560?, 0x140000db6e8?}, {0x60?, ...})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:715 +0x190
encoding/json.ptrEncoder.encode({0x140000db7d8?}, 0x14000272580, {0x101473260?, 0x140001bf560?, 0x0?}, {0x20?, 0xb5?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:887 +0x1dc
encoding/json.(*encodeState).reflectValue(0x14000272580, {0x101473260?, 0x140001bf560?, 0x1?}, {0x8?, 0x0?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:322 +0x70
encoding/json.(*encodeState).marshal(0x1400004d978?, {0x101473260?, 0x140001bf560?}, {0x84?, 0xd?})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:298 +0xb8
encoding/json.Marshal({0x101473260, 0x140001bf560})
        /Users/knight/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.darwin-arm64/src/encoding/json/encode.go:164 +0x8c
github.com/privateerproj/privateer-sdk/pluginkit.(*Vessel).WriteResults(0x140001bf560)
        /Users/knight/go/pkg/mod/github.com/privateerproj/privateer-sdk@v1.2.0/pluginkit/vessel.go:104 +0x90
github.com/privateerproj/privateer-sdk/pluginkit.(*Vessel).Mobilize(0x140001bf560)
        /Users/knight/go/pkg/mod/github.com/privateerproj/privateer-sdk@v1.2.0/pluginkit/vessel.go:95 +0x298
github.com/privateerproj/privateer-sdk/command.NewPluginCommands.debugCommand.func1(0x14000219208, {0x1011c6d30?, 0x4?, 0x1011c6d4c?})
        /Users/knight/go/pkg/mod/github.com/privateerproj/privateer-sdk@v1.2.0/command/plugin.go:63 +0x50
github.com/spf13/cobra.(*Command).execute(0x14000219208, {0x140001e5730, 0x1, 0x1})
        /Users/knight/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0x814
github.com/spf13/cobra.(*Command).ExecuteC(0x14000218f08)
        /Users/knight/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(0x1011cd21f?)
        /Users/knight/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071 +0x1c
main.main()
        /Users/knight/dev/revanite/pvtr-github-repo/main.go:52 +0x32c
exit status 2

</details>

--- a/data/graphql-data.go
+++ b/data/graphql-data.go
@@ -21,9 +21,9 @@ type GraphqlRepoData struct {
 		Object struct {
 			Tree struct {
 				Entries []struct {
-					Name   string
-					Type   string // "blob" for files, "tree" for directories
-					Path   string
+					Name string
+					Type string // "blob" for files, "tree" for directories
+					Path string
 				}
 			} `graphql:"... on Tree"`
 		} `graphql:"object(expression: \"HEAD:\")"`
@@ -87,7 +87,7 @@ type GraphqlRepoData struct {
 		}
 		ContributingGuidelines struct {
 			Body         string
-			ResourcePath githubv4.URI
+			ResourcePath string
 		}
 		DependencyGraphManifests struct {
 			TotalCount int
@@ -96,8 +96,8 @@ type GraphqlRepoData struct {
 				Dependencies struct {
 					TotalCount int
 					Nodes      []struct {
-						PackageName    string
-						Requirements   string
+						PackageName  string
+						Requirements string
 					}
 				} `graphql:"dependencies(first: 100)"`
 			} `graphql:"nodes"`

